packages/ui/src/components/va-virtual-scroller/useVirtualScrollerSizes.ts

```diff
@@ -0,0 +1,78 @@
+import { PropType, ExtractPropTypes, ref, Ref, shallowRef, computed, watch, onMounted } from 'vue'
+
+import { useResizeObserver } from '../../composables'
+
+export const useVirtualScrollerSizesProps = {
+  horizontal: { type: Boolean, default: false },
+  itemSize: {
+    type: [Number, String] as PropType<string | number>,
+    default: 0,
+    validator: (v: number | string) => {
+      if (typeof v === 'string') { return (!isNaN(+v) || v.endsWith('px') || v.endsWith('rem')) && parseInt(v) >= 0 }
```

Comment:
Maybe we can provide more descriptive error messages via `warn` function we have in utils?

validator won't tell much of what's wrong